sycl/include/sycl/ext/oneapi/experimental/invoke_simd.hpp

```diff
@@ -420,6 +431,7 @@ __attribute__((always_inline)) auto invoke_simd(sycl::sub_group sg,
   // what the subgroup size is and arguments don't need widening and return
   // value does not need shrinking by this library or SPMD compiler, so 0
   // is fine in this case.
+  detail::verify_valid_args<T...>();
   detail::verify_no_ref<Callable>();
```

Comment:
The name of the function seems a bit too general. Now it checks only that all arguments are trivially_copiable.
If keep the name, then it would be reasonable to move checks from verify_no_ref to that function as well.

Reply:
good point, im gonna pick the second option so it is more extensible in the future, thanks